IdentityCore/tests/integration/MSIDDefaultTokenCacheIntegrationTests.m

```diff
@@ -0,0 +1,604 @@
+{
+    MSIDAccount *account = [[MSIDAccount alloc] initWithUpn:DEFAULT_TEST_ID_TOKEN_USERNAME
+                                                       utid:DEFAULT_TEST_UTID
+                                                        uid:DEFAULT_TEST_UID];
+    
+    NSError *error = nil;
+    MSIDToken *token = [_cacheAccessor getSharedRTForAccount:account
+                                               requestParams:[MSIDTestRequestParams v2DefaultParams]
+                                                     context:nil
+                                                       error:&error];
+    
+    XCTAssertNil(error);
+    XCTAssertNil(token);
+}
+
+- (void)testGetSharedRTForAccount_whenAccountWithUtidAndUidProvided_shouldReturnToken
+{
+    MSIDToken *token = [[MSIDToken alloc] initWithTokenResponse:[MSIDTestTokenResponse v2DefaultTokenResponse]
+                                                        request:[MSIDTestRequestParams v2DefaultParams]
+                                                      tokenType:MSIDTokenTypeRefreshToken];
+    
+    MSIDAccount *account = [[MSIDAccount alloc] initWithUpn:nil
+                                                       utid:DEFAULT_TEST_UTID
+                                                        uid:DEFAULT_TEST_UID];
+    
+    [_cacheAccessor saveSharedRTForAccount:account
+                              refreshToken:token
+                                   context:nil
+                                     error:nil];
+    
+    NSError *error = nil;
+
+    MSIDToken *returnedToken = [_cacheAccessor getSharedRTForAccount:account
+                                                        requestParams:[MSIDTestRequestParams v2DefaultParams]
+                                                              context:nil
+                                                                error:&error];
+    
+    XCTAssertNil(error);
+    XCTAssertNotNil(returnedToken);
+    XCTAssertEqualObjects(token, returnedToken);
+}
+
+- (void)testGetSharedRTForAccount_whenAccountWithUtidAndUidProvided_andOnlyAT_shouldReturnNil
+{
+    MSIDToken *token = [[MSIDToken alloc] initWithTokenResponse:[MSIDTestTokenResponse v2DefaultTokenResponse]
+                                                        request:[MSIDTestRequestParams v2DefaultParams]
+                                                      tokenType:MSIDTokenTypeAccessToken];
+    
+    MSIDAccount *account = [[MSIDAccount alloc] initWithUpn:nil
+                                                       utid:DEFAULT_TEST_UTID
+                                                        uid:DEFAULT_TEST_UID];
+    
+    [_cacheAccessor saveSharedRTForAccount:account
+                              refreshToken:token
+                                   context:nil
+                                     error:nil];
+    
+    NSError *error = nil;
+
+    MSIDToken *returnedToken = [_cacheAccessor getSharedRTForAccount:account
```

Comment:
I'm a bit confused about this test as it's saving a refresh token and then trying to get it back. Why is it expecting nil?

Reply:
it should've been saveAT.